packages/components/src/PieChartButton/PieChartButton.component.js

```diff
@@ -0,0 +1,421 @@
+	return percentage * 2 / 100 * Math.PI;
+}
+
+/**
+ * this function generates chart's empty part
+ * @param {array} values the values shown in the graph
+ * @param {object} size the current size
+ * @param {function} arcGen the arc generator
+ * @param {number} minimumPercentage the minimum percentage to be shown
+ */
+export function getEmptyPartCircle(values, size, minimumPercentage) {
+	const allPercentages = values.reduce((acc, value) => acc + value.percentageShown, 0);
+	if (allPercentages >= MAX_PERCENT - minimumPercentage) {
+		return null;
+	}
+
+	const arcGenerated = arcGen({
+		innerRadius: size.innerRadius,
+		outerRadius: size.outerRadius,
+		padAngle: size.padAngle,
+		startAngle: getAngle(allPercentages),
+		endAngle: Math.PI * 2,
+	});
+	return (
+		<path
+			className={classnames(theme['tc-pie-chart-color-alto'], 'tc-pie-chart-color-alto')}
+			d={arcGenerated}
+			transform={`translate(${size.svgSize / 2},${size.svgSize / 2})`}
+		/>
+	);
+}
+
+/**
+ * This function return the sum of the percentages below the given index
+ * @param {array} values the values to get when we have to start
+ * @param {number} index the current index
+ */
+export function getPercentageToIndex(values, index) {
+	return values.reduce((acc, value, i) => {
+		if (i < index) {
+			return acc + value.percentageShown;
+		}
+		return acc;
+	}, 0);
+}
+
+/**
+ * This function generate a part of circle for a value
+ * @param {object} value the value to generate
+ * @param {number} index the current index
+ * @param {array} values the values to get when we have to start
+ * @param {object} size the current graph size
+ */
+export function getCircle(value, index, values, size) {
+	const percentagesDone = getPercentageToIndex(values, index);
+
+	const arcGenerated = arcGen({
+		innerRadius: size.innerRadius,
+		outerRadius: size.outerRadius,
+		padAngle: size.padAngle,
+		startAngle: getAngle(percentagesDone),
+		endAngle: getAngle(percentagesDone + value.percentageShown),
+	});
+
+	return (
+		<path
+			key={index}
+			d={arcGenerated}
+			className={classnames(
+				theme[`tc-pie-chart-color-${value.color}`],
+				`tc-pie-chart-color-${value.color}`,
+			)}
+			transform={`translate(${size.svgSize / 2},${size.svgSize / 2})`}
+		/>
+	);
+}
+
+/**
+ * This function is just a sort
+ * @param {object} a an element with a percentageShown to compare
+ * @param {object} b an other element with a percentageShown to compare
+ */
+function sortElements(a, b) {
+	if (a.percentageShown > b.percentageShown) {
+		return -1;
+	}
+	if (a.percentageShown < b.percentageShown) {
+		return 1;
+	}
+	return 0;
+}
+
+/**
+ * This function sets minimum percentage show with the minimum percentage shown
+ * It increase values below minPercentage & decrease values above if some are increased
+ * @param {array} values the set of values
+ * @param {number} minimumPercentage the minimum value we have to show
+ */
+export function setMinimumPercentage(model, minimumPercentage) {
```

Comment:
could you add an example on the jsdoc to better to understand what do this method.

setMinimumPercentage([{percentage: 50}, {percentage: 2}], 5) => [{percentage: 50, percentageShown: 47}, {percentage: 2, percentageShown: 5}]